fastlane/Fastfile

```diff
@@ -109,6 +109,44 @@ platform :ios do
     )
   end
   
+  desc "Upload a local IPA to Firebase App Distribution with (app_id) and (group)."
+  desc "#### Options"
+  desc " * **`generate_changelog`**: Whether or not to generate a changelog as the release notes (boolean – default: `false`)"
+  desc " * **`changelog_type`**: The type of changelog to generate (e.g. `\"git\"`, `\"jenkins\"`, `\"pr\"` – default: `\"git\"`)"
+  desc " * **`release_notes`**: A string to set as the release notes; overrides any generated changelog."
+  lane :uploadToFirebase do |options|
+    options[:generate_changelog] ||= false
+    options[:changelog_type] ||= "git"
+
+    raise "uploadToFirebase: An app ID must be passed as an option (e.g. `\"1:123456789:ios:abcd1234\"`)." unless options[:app_id]
+    raise "uploadToFirebase: A tester group must be passed as an option." unless options[:group]
+
+    unless options.key?(:release_notes)
+      if options[:generate_changelog] == true
+        case options[:changelog_type]
+          when "jenkins"
+            make_changelog_from_jenkins
+
+          when "pr"
+            UI.error("uploadToCrashlytics: Environment variable for PR ID is not set (ghprbPullId).") unless ENV.key?("ghprbPullId")
+            options[:release_notes] = "PR: ##{ENV["ghprbPullId"]} (#{ENV["GIT_BRANCH"]})"
+
+          when "git"
+            changelog_from_git_commits
+
+          else
+            UI.error("Unrecognized changelog_type: #{options[:changelog_type]}")
+        end
+      end
+    end
+
+    firebase_app_distribution(
+      app: options[:app_id],
+      release_notes: options[:release_notes], # if release_notes is nil, this will be overriden by any generated changelog.
```

Comment:
I'm not completely sure how this works or if the release notes override would work for the `firebase_app_distribution` block if this is `nil`. Is this something that is unique to the `crashlytics` block or could it work here, too?